pkg/api/v1/validation/validation.go

```diff
@@ -78,7 +78,7 @@ func validateContainerResourceName(value string, fldPath *field.Path) field.Erro
 			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
 		}
 	}
-	return field.ErrorList{}
+	return allErrs
```

Comment:
In fact, all errors are returned in the above return(s). But here both are ok, return allErrs might be safer.

Reply:
yes, by the way, there are many validations, which do not return until all fields are validated.  (https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/validation/validation.go#L131) etc.
should we return err ASAP or after all validated. To improve apiserver performance, I think return err ASAP is better.

Reply:
I do not think all errs are returned.  validateResourceName(...) may return errs but when `if len(strings.Split(value, "/")) != 1` , nothing will be returned.  That is not expected.

Reply:
I sent an issue and pr for this before.  But do not consider v1 package.
https://github.com/kubernetes/kubernetes/pull/51257
https://github.com/kubernetes/kubernetes/issues/51256

Reply:
original code 
```
func validateContainerResourceName(value string, fldPath *field.Path) field.ErrorList {
	allErrs := validateResourceName(value, fldPath)
	if len(strings.Split(value, "/")) == 1 {
		if !helper.IsStandardContainerResourceName(value) {
			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
		}
	}
	return field.ErrorList{}
}
```
as @NickrenREN  say, when ```len(strings.Split(value, "/")) != 1```  may return wrong result.

Reply:
Good point! I misread :)